drivers/mtd_spi_nor/mtd_spi_nor.c

```diff
@@ -622,19 +659,50 @@ static int mtd_spi_nor_write_page(mtd_dev_t *mtd, const void *src, uint32_t page
     /* write enable */
     mtd_spi_cmd(dev, dev->params->opcode->wren);
 
+    /* Wait for WEL-Flag to be set */
+    wait_for_write_enable_set(dev);
```

Comment:
Is this required for all chips / when integrity checking is not enabled?

Reply:
Yes, on all Flash devices you have to wait for the WEL flag to be set before starting a Write/Erase operation. Otherwise the operation will be ignored.

https://www.macronix.com/Lists/Datasheet/Attachments/8868/MX25R6435F,%20Wide%20Range,%2064Mb,%20v1.6.pdf Page 34 has a block diagram. However now that I look at it, Macronix says to issue the WREN command again if the flag is not set. So the right thing to do would not be to wait for the flag to be set but trying to set it until it is set.

Maybe that would need a try-counter instead of a timeout. :thinking: